tests/flow/test_optimizations_plan.py

```diff
@@ -426,3 +426,21 @@ def test27_optimize_cartesian_product_complex_filter_trees(self):
                     [0, 3],
                     [0, 3]]
         self.env.assertEqual(resultset, expected)
+
+    def test28_optimize_label_scan_cached_label_id(self):
```

Comment:
Consider flushing / deleting the graph prior to execution, to create a "pure" baseline.
Please add a few comments explaining the purpose of this test,
Try to perform additional validations on the queries making sure the generated plan is as expected.